### DISCUSSION TITLE
How much space will Homebrew take up on my M1 computer?

### DISCUSSION THREAD

**Discussion by rorynl** [Homebrew/discussions#3325] [Getting started]
Hi,

I'd like to know how many megabytes or gigabytes of space Homebrew takes up on initial install. I can take into account Xcode Command Line Tools separately. If anybody has this info for MacPorts and/or Nix Package Manager, that would be great too.

Unless I'm missing something, this information does not appear to be readily available.

Thanks

**Comment by SMillerDev** [comment]
After cloning the `homebrew/brew` repo the system reports that it uses 82MB. That is the most minimal version of the install you could configure (without installing anything).

Most people on macOS will also appreciate being able to install software from the repos though, which are:
`homebrew/core`: 556MB
`homebrew/cask`: 354MB

Overall this would take just under 0.4% of the storage of the 256GB MacBook Air.

**Reply by carlocab** [reply]
You can also use the (currently experimental) `HOMEBREW_INSTALL_FROM_API` environment variable that will allow you to use Homebrew without having to clone `homebrew/core` or `homebrew/cask`.

**Reply by bassamsdata** [reply]
How can I don HOMEBREW_INSTALL_FROM_API set to true. is there any step by step tutorial. I searched for tutorials but didn't find any.

**Reply by kresdjan** [reply]
`export HOMEBREW_INSTALL_FROM_API=1`

https://docs.brew.sh/Installation#default-tap-cloning

**Reply by carlocab** [reply]
`HOMEBREW_INSTALL_FROM_API` is now the default. There is no need to set it. You may need to
```
unset HOMEBREW_NO_INSTALL_FROM_API
```
if you have it set.
